tests/publish-dsl.nf

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env nextflow
+/*
+ * Copyright 2013-2024, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+nextflow.preview.topic = true
+
+process align {
+  input:
+  val(x)
+
+  output:
+  path("*.bam")
+  path("${x}.bai")
+
+  """
+  echo ${x} > ${x}.bam
+  echo ${x} | rev > ${x}.bai
+  """
+}
+
+process my_combine {
+  input:
+  path(bamfile)
+  path(baifile)
+
+  output:
+  path 'result.txt'
+
+  """
+  cat $bamfile > result.txt
+  cat $baifile >> result.txt
+  """
+}
+
+process foo {
+  output:
+  path 'xxx'
+
+  '''
+  mkdir xxx
+  touch xxx/A
+  touch xxx/B
+  touch xxx/C
+  '''
+}
+
+workflow {
+  def input = Channel.of('alpha','beta','delta')
+  align(input)
+
+  def bam = align.out[0].toSortedList { it.name }
+  def bai = align.out[1].toSortedList { it.name }
+  my_combine( bam, bai )
+  my_combine.out.view{ it.text }
+
+  foo | topic('foo')
```

Comment:
I would support a `topic:` block as it's a nice way to give some structure to how topics are used.

What if the user wants to send multiple channels to the same topic? Simplest thing would be for them to use `mix`:

```groovy
topic:
foobar = foo.out | mix( bar.out )
```

Reply:
It think also repeting it, should be fine e.g .

```
topic:
foobar = foo.out 
foobar = bar.out
```

Reply:
I was thinking the same, though it would be best to avoid a proliferation of different ways to write the same thing. Maybe we can emphasize only one way in the docs

Reply:
I'm more scared about the fact that it's commonly assumed the latter assignment, override the former one.

Reply:
Yes that would be confusing for many people

I've always thought of topics as a push-based approach, whereas assignment is pull-based. Assignment doesn't fit very well when you can send multiple things to the same topic